master/buildbot/locks.py

```diff
@@ -320,19 +328,23 @@ class LockAccess(util.ComparableMixin):
 
     @param mode: Mode of accessing the lock.
     @type  mode: A string, either 'counting' or 'exclusive'.
+
+    @param count: How many units does the access occupy
+    @type  count: Integer, not negative, default is 1 for backwards
+                  compatibility
     """
 
-    compare_attrs = ('lockid', 'mode')
+    compare_attrs = ('lockid', 'mode', 'count')
 
-    def __init__(self, lockid, mode, _skipChecks=False):
+    def __init__(self, lockid, mode, count=1):
         self.lockid = lockid
         self.mode = mode
+        self.count = count
 
-        if not _skipChecks:
-            # these checks fail with mock < 0.8.0 when lockid is a Mock
-            # TODO: remove this in Buildbot-0.9.0+
-            assert isinstance(lockid, (MasterLock, WorkerLock))
-            assert mode in ['counting', 'exclusive']
+        assert isinstance(lockid, (MasterLock, WorkerLock))
+        assert mode in ['counting', 'exclusive']
+        assert isinstance(count, int)
+        assert count >= 0
```

Comment:
I think it makes sense to not allow any other lock count than 1 for exclusive access. Exclusive access means that a single lock access can acquire the lock, so the lock count does not change the behavior. Best to throw exception if count != 1 in exclusive mode.